tests/clusters/test_cluster.py

```diff
@@ -80,6 +80,28 @@
 ]
 
 
+TEST_SLICE_CONFIG = [
+    {
+        "host": "host_slice",
+        "port": 9000,
+        "user": "default",
+        "password": "",
+        "database": "slice_1_default",
+        "http_port": 8123,
+        "storage_sets": {"generic_metrics_distributions_1"},
+        "single_node": True,
+    }
+]
+
+
+OVERRIDE_STORAGE_SET_KEY_CONFIG = {
+    "generic_metrics_distributions": {
+        0: "generic_metrics_distributions",
+        1: "generic_metrics_distributions_1",
+    }
+}
```

Comment:
I don't really understand how this pieces fit together. How is 0 and 1 defined? I would prefer if we do not attempt to override storage sets at all and just map storage set/partition pairs which are unique to a cluster. I feel it is unnecessary and adds additional complexity.

Reply:
I'm not sure I understand this comment. 0 and 1 are physical partitions and so she is using a storage set key+partition pair to do this override.

So are you saying that CLUSTERS should contain partition information? My thought is that should not be necessary for unsliced storages which are the vast majority, and will be completely unused by single-tenant/self-hosted users

Reply:
Would you have the same opposition if this was written out (and documented) as:

```python
STORAGE_SET_KEY_PARTITION_MAP = {
     "storage_set_key_no_partitions": {
           part_number_0: "storage_set_key_partition_0",
           # ...
           part_number_n: "storage_set_key_partition_n"
     }
}
```

Reply:
Yes, to summarize my point is that this is currently mapping:

storage set / partition pair -> another thing that we are also kind of calling a storage set but is not really -> cluster

Why not just map:
storage set / partition pair (which is already unique) -> cluster

My suggestion is to remove the part in the middle and go straight from (storage set / partition) to cluster. The extra complexity is both in Snuba code as well as partially exposed to users via these settings which is why I am opposed to it.

Reply:
I don't agree with "another thing that we are also kind of calling a storage set but is not really". My interpretation is that `storage_set_key_partition_n` is a storage set as it defines colocated resources to clusters (as defined by settings.CLUSTERS) for slice n. It is perhaps not referred to directly in the storage definition, but that's fine with me.

If we changed CLUSTERS to contain partition information, then I think we're adding extra complexity to cluster definitions and parsing from which almost every dataset and almost every user derives no value. Therefore I still think we'd probably want to create an override structure with your approach.

If we create that cluster override structure, then we end up with a mapping like:

```python
PARTITIONED_CLUSTERS = [
   {
        "host": "host_slice",
        "port": 9000,
        "user": "default",
        "password": "",
        "database": "default",
        "http_port": 8123,
        "storage_sets": {("generic_metrics_distributions",0)},
        "single_node": True,
   },
   {
        "host": "host_slice",
        "port": 9001,
        "user": "default",
        "password": "",
        "database": "default",
        "http_port": 8124,
        "storage_sets": {("generic_metrics_distributions",1)},
        "single_node": True,
   },
]
```

which in my opinion is the same complexity to support in Snuba as what Riya did. I'm going to write up a bit more in notion but unless you think that this is functionally limiting I don't think it should block this approach

Reply:
This example is more straightforward to me that the one in the code with the storage set overrides since there is no need to write `storage_set_key_partition_n` in two different places which was the additional complexity I was referring to (to clarify I think this is more mental complexity of having another thing here more than overall code complexity which is likely on par either way). You could leave CLUSTERS unchanged and only put the overrides in a structure like this.